core/native/refresh/src/main/csharp/ViewModels/Info/VersionsViewModel.cs

```diff
@@ -0,0 +1,204 @@
+
+namespace Ch.Cyberduck.Core.Refresh.ViewModels.Info
+{
+    public class VersionsViewModel : ReactiveObject
+    {
+        private readonly ObservableAsPropertyHelper<VersionViewModel> selectedVersionProperty;
+        private readonly SourceList<VersionModel> versions = new();
+        private readonly IObservableCache<VersionViewModel, VersionModel> viewModelCache;
+
+        public VersionsViewModel(Controller controller, SessionPool session)
+        {
+            var temporary = TemporaryFileServiceFactory.get();
+            var versioning = (Versioning)session.getFeature(typeof(Versioning));
+
+            /* setup tracking */
+            viewModelCache = versions.Connect()
+                .AddKey(x => x)
+                .Transform(x => new VersionViewModel(x))
+                .Bind(Versions)
+                .AsObservableCache();
+
+            this.WhenAnyValue(x => x.SelectedVersionValue)
+                .Select(x => x switch
+                {
+                    null => Observable.Return(default(VersionViewModel)),
+                    _ => viewModelCache.WatchValue(x)
+                })
+                .Switch().ToProperty(this, nameof(SelectedVersion), out selectedVersionProperty);
+
+            /* setup commands */
+            Open = ReactiveCommand.Create(() =>
+            {
+                var f = SelectedVersionValue.Path;
+                controller.background(new QuicklookTransferBackgroundAction(
+                    controller, QuickLookFactory.get(), session, Collections.singletonList(
+                        new TransferItem(f, temporary.create(session.getHost().getUuid(), f)))));
+            }, this.WhenAnyValue(v => v.SelectedVersionValue).Select(v => v != null));
+            Remove = ReactiveCommand.CreateFromTask(async () =>
+            {
+                var norm = PathNormalizer.normalize(Collections.singletonList(SelectedVersionValue.Path));
+                if (norm.size() == 0)
+                {
+                    return;
+                }
+                var native = Utils.ConvertFromJavaList<Path>(norm);
+                if (!await PromptDelete.Handle(native))
+                {
+                    return;
+                }
+                try
+                {
+                    Busy = true;
+                    TaskCompletionSource<object> result = new();
+                    controller.background(
+                        new AsyncWorkerBackgroundAction(controller, session, result,
+                            new DeleteWorker(
+                                LoginCallbackFactory.get(controller), norm,
+                                PathCache.empty(), new DisabledProgressListener())));
+                    await result.Task;
+                }
+                finally
+                {
+                    Busy = false;
+                }
+                await Load.ExecuteIfPossible();
+            }, this.WhenAnyValue(v => v.SelectedVersionValue).Select(v => v != null));
```

Comment:
Shouldn't we check whether Delete is possible?
i.e. OneDrive cannot delete single versions.